parquet_file/src/catalog.rs

```diff
@@ -377,6 +388,36 @@ fn parse_uuid_required(s: &str) -> Result<Uuid> {
     parse_uuid(s)?.context(UuidRequired {})
 }
 
+/// Parse [`DirsAndFilename`](object_store::path::parsed::DirsAndFileName) from protobuf.
+fn parse_dirs_and_filename(proto: &Option<proto::Path>) -> Result<DirsAndFileName> {
+    match proto {
+        Some(proto) => Ok(DirsAndFileName {
+            directories: proto
+                .directories
+                .iter()
+                .map(|s| PathPart::from(&s[..]))
+                .collect(),
+            file_name: Some(PathPart::from(&proto.file_name[..])),
+        }),
+        None => Err(Error::PathRequired {}),
+    }
```

Comment:
This could be:

```suggestion
    let proto = proto.as_ref().context(PathRequired)?;

    Ok(DirsAndFileName {
        directories: proto
            .directories
            .iter()
            .map(|s| PathPart::from(&s[..]))
            .collect(),
        file_name: Some(PathPart::from(&proto.file_name[..])),
    })
```

Not sure which you find clearer, I can see arguments either way :)